src/bin/pg_autoctl/monitor.c

```diff
@@ -1190,3 +1199,261 @@ monitor_get_notifications(Monitor *monitor)
 
 	return true;
 }
+
+
+/*
+ * monitor_get_extension_version gets the current extension version from the
+ * Monitor's Postgres catalog pg_available_extensions.
+ */
+bool
+monitor_get_extension_version(Monitor *monitor, MonitorExtensionVersion *version)
+{
+	MonitorExtensionVersionParseContext context = { version, false };
+	PGSQL *pgsql = &monitor->pgsql;
+	const char *sql =
+		"SELECT default_version, installed_version"
+		"  FROM pg_available_extensions WHERE name = $1";
+	int paramCount = 1;
+	Oid paramTypes[1] = { TEXTOID };
+	const char *paramValues[1];
+
+	paramValues[0] = PG_AUTOCTL_MONITOR_EXTENSION_NAME;
+
+	if (!pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
+								   &context, &parseExtensionVersion))
+	{
+		log_error("Failed to get the current version for extension \"%s\", "
+				  "see previous lines for details.",
+				  PG_AUTOCTL_MONITOR_EXTENSION_NAME);
+		return false;
+	}
+
+	if (!context.parsedOK)
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	/* disconnect from PostgreSQL now */
+	pgsql_finish(&monitor->pgsql);
+
+	return true;
+}
+
+
+/*
+ * parseExtensionVersion parses the resultset of a query on the Postgres
+ * pg_available_extension_versions catalogs.
+ */
+static void
+parseExtensionVersion(void *ctx, PGresult *result)
+{
+	MonitorExtensionVersionParseContext *context =
+		(MonitorExtensionVersionParseContext *) ctx;
+
+	char *value = NULL;
+	int length;
+
+	/* we have rows: we accept only one */
+	if (PQntuples(result) != 1)
+	{
+		log_error("Query returned %d rows, expected 1", PQntuples(result));
+		context->parsedOK = false;
+		return;
+	}
+
+	if (PQnfields(result) != 2)
+	{
+		log_error("Query returned %d columns, expected 2", PQnfields(result));
+		context->parsedOK = false;
+		return;
+	}
+
+	if (PQgetisnull(result, 0, 0) || PQgetisnull(result, 0, 1))
+	{
+		log_error("pgautofailover default_version or installed_version "
```

Comment:
can we use `PG_AUTOCTL_MONITOR_EXTENSION_NAME` instead of hardcoding the name of the extension?

Reply:
Fixed.